rust/kernel/src/printk.rs

```diff
@@ -52,21 +57,51 @@ impl fmt::Write for LogLineWriter {
 
 /// [`println!`] functions the same as it does in `std`, except instead of
 /// printing to `stdout`, it writes to the kernel console at the `KERN_INFO`
-/// level.
+/// level. Level can be overwritten with the argument level.
 ///
 /// [`println!`]: https://doc.rust-lang.org/stable/std/macro.println.html
 #[macro_export]
 macro_rules! println {
     () => ({
-        $crate::printk::printk("\n".as_bytes());
+        println!(level: $crate::printk::KERN_INFO);
+    });
+    (level: $level:expr) => ({
+        $crate::printk::printk("\n".as_bytes(), $level);
+    });
+    ($msg:expr) => ({
+        println!(level: $crate::printk::KERN_INFO, $msg);
+    });
+    (level: $level:expr, $msg:expr) => ({
+        $crate::printk::printk(concat!($msg, "\n").as_bytes(), $level);
+    });
+    (level: $level:expr, $fmt:expr, $($arg:tt)*) => ({
```

Comment:
I'm not wild about this syntax. I think we should just add `eprintln!()` rather than add an API that's subtly different from the `println!()` everyone knows.

If we want to add our own API to provide this (e.g. `kprintln!()`) that'd be ok with me though.

Reply:
I added `kprintln!` with for that feature.
(It's still untested as I can't build this kernel)